allennlp/training/__init__.py

```diff
@@ -1,2 +1,3 @@
 from allennlp.training.model import Model
+from allennlp.training.trainer import Trainer
```

Comment:
You switched to `TrainDriver`, so this should just be removed.  Except, as I said above, I think it'd be better to just stick with `Trainer`, and move everything there, and get rid of the notion of "drivers" entirely.